### PR TITLE
transformNumeric: Handle "0" as the explicit value

### LINKED ISSUE #100: Using "0" (zero) as a numeric prop value results into no CSS
## What:

```jsx
<Composition margin={0} />
```

Results into no `margin` CSS property being output into CSS. The same relates to any other numeric property that has `0` as its value.

## Why:

`0` is treated as a fasly value and produces no expected CSS property value.

## Environment:

- atomic-layout version: `0.6.0`

## Expected behavior:

Zero as a value results into `0` as a value of the respective CSS property: `margin: 0;`

### PR DESCRIPTION
**Pull request by ruhaise**
## Change log
- Fixes #100 
- [x] `transformNumeric` now asserts if the `value` is set, not if it's truthy
- [x] Fix a typo in existing `transformNumeric` unit test
- [x] Add unit tests to cover the handling of "0" as a numeric value
- [x] Add integration test for `margin: 0` scenario

### CODE REVIEW

File: src/utils/math/transformNumeric.spec.ts [hunk 1]
```diff
@@ -11,3 +11,7 @@ test('Returns string values as is', () => {
 test('Returns "undefined" when no value is given', () => {
   expect(transformNumeric()).toEqual('')
 })
+
+test('Suffixes numeric values with "px" string, if value is zero the result should be 0px', () => {
```
Comment (kettanaito) [inline src/utils/math/transformNumeric.spec.ts]: Could you please rename this test suit to `Outputs CSS for 0 as a value`.

File: src/utils/math/transformNumeric.spec.ts [hunk 2]
```diff
@@ -11,3 +11,7 @@ test('Returns string values as is', () => {
...
+  expect(transformNumeric(0)).toEqual('0px')
```
Comment (kettanaito) [inline src/utils/math/transformNumeric.spec.ts]: Let's also assert if we pass `'0'`:

```js
expect(transformNumeric('0')).toEqual('0')
```

Comment (ruhaise) [reply src/utils/math/transformNumeric.spec.ts]: renamed the test suite

File: src/utils/math/transformNumeric.spec.ts (hunk 1 above)
Comment (ruhaise) [reply src/utils/math/transformNumeric.spec.ts]: renamed test suite

File: stories/Demo.jsx [hunk 3]
```diff
@@ -29,7 +29,8 @@ const templateTv = `
 const Demo = () => (
   <div>
     <Composition
-      gutter={16}
+      gutter={0} // this is not getting generated
```
Comment (kettanaito) [inline stories/Demo.jsx]: I need to verify this behavior on my local.

**Comment by kettanaito** [comment]
@ruhaise, do you think it's reasonable to suffix `0`?

There is no substantial difference between `0px`, `0rem` or `0%`. Seems like we can get rid of the suffix in case of zero as a value, and save from 1 to 3 unnecessary characters in the compiled CSS.

**Comment by ruhaise** [comment]
> @ruhaise, do you think it's reasonable to suffix `0`?
> 
> There is no substantial difference between `0px`, `0rem` or `0%`. Seems like we can get rid of the suffix in case of zero as a value, and save from 1 to 3 unnecessary characters in the compiled CSS.

@kettanaito  you are right, its good to consider zero as a special case and not to attach any suffix to it

File: src/utils/math/transformNumeric.spec.ts [hunk 4]
```diff
@@ -4,20 +4,15 @@ test('Suffixes numeric values with "px" string', () => {
   expect(transformNumeric(5)).toEqual('5px')
 })
 
-test('Returns string values as is', () => {
+test('Returns string values as it is', () => {
```
Comment (kettanaito) [inline src/utils/math/transformNumeric.spec.ts]: There is nothing wrong with "as is" grammar construction ;)

Comment (ruhaise) [reply src/utils/math/transformNumeric.spec.ts]: my bad

Comment (ruhaise) [reply src/utils/math/transformNumeric.spec.ts]: reverted

**Comment by ruhaise** [comment]
> The current unit tests and `transformNumeric` is great. I need to add integration test for `margin: 0` as well.

im not able to run integration test at all, it gives some weird error
image

do you know why is this ? i didnt chnage anything else on my local

**Comment by kettanaito** [comment]
@ruhaise, that looks weird.

I experience a similar behavior in `react-advanced-form` running Cypress. This may be a Cypress bug, at the latest. Try removing `node_modules` and installing them once more (preferably via `yarn`).

**Comment by ruhaise** [comment]
> @ruhaise, that looks weird.
> 
> I experience a similar behavior in `react-advanced-form` running Cypress. This may be a Cypress bug, at the latest. Try removing `node_modules` and installing them once more (preferably via `yarn`).

tried that no luck, are experiencing same

**Comment by kettanaito** [comment]
@ruhaise, could you please try to execute the following commands:

```bash
yarn run test:e2e
```

**Comment by ruhaise** [comment]
> @ruhaise, could you please try to execute the following commands:
> 
> ```shell
> yarn run test:e2e
> ```

i have already tried that before, only i can run unit test by `test:unit`, :(

**Comment by kettanaito** [comment]
@ruhaise I've got the integration tests working on my local. See the instructions below.

```bash
git checkout 100-zero-value-bug
git pull --rebase
rm -rf node_modules
yarn install
```

After that running `yarn run test:e2e` (or `yarn run cypress`) should launch the testing client. Let me know if that helps.

**Comment by ruhaise** [comment]
> @ruhaise I've got the integration tests working on my local. See the instructions below. You would need to update your forked branch, since I've just merged an updated master in it.
> 
> ```shell
> git checkout 100-zero-value-bug
> git pull --rebase
> rm -rf node_modules
> yarn install
> ```
> 
> After that running `yarn run test:e2e` (or `yarn run cypress`) should launch the testing client. Let me know if that helps.

@kettanaito  its working again, its weird since there is no changes when you have merged master to this brach,

File: cypress/integration/other/prop-aliases.spec.js [hunk 5]
```diff
@@ -9,6 +9,7 @@ const assertValues = {
   colStart: '2',
   colEnd: '3',
   gutter: '20px 25px',
+  margin: 0,
```
Comment (kettanaito) [inline cypress/integration/other/prop-aliases.spec.js]: This now tests `margin:0` scenario, but doesn't test positive value (i.e. `margin:10px`).
Hm, we may want to come up with another approach here, these overrides are not designed to handle multiple value assertions for a single prop alias.

**Comment by kettanaito** [comment]
Added support for multiple values assertions (c2064f14eeb50069c614530703fdb9a027b98017). Now the integration test for `margin` should assert both zero and positive values.

**Comment by kettanaito** [comment]
Congratulate with your first contribution! 🎉

**Comment by redraushan** [comment]
Congratulations ruhaise for the kickstart!

**Pull request merged**